openbas-front/src/admin/components/common/injects/expectations/InjectAddExpectation.tsx

```diff
@@ -41,8 +41,18 @@ const InjectAddExpectation: FunctionComponent<InjectAddExpectationProps> = ({
   const handleClose = () => setOpenDialog(false);
 
   // Form
-  const onSubmit = (data: ExpectationInput) => {
-    handleAddExpectation(data);
+  const onSubmit = (data: ExpectationInputForm) => {
+    const values: ExpectationInput = {
+      expectation_type: data.expectation_type,
+      expectation_name: data.expectation_name,
+      expectation_description: data.expectation_description,
+      expectation_score: data.expectation_score,
+      expectation_expectation_group: data.expectation_expectation_group,
+      expectation_expiration_time: data.expiration_time_days * 3600 * 24
+        + data.expiration_time_hours * 3600
+        + data.expiration_time_minutes * 60,
```

Comment:
I think you can do that:
{
     ...data,
      expectation_expiration_time: data.expiration_time_days * 3600 * 24
        + data.expiration_time_hours * 3600
        + data.expiration_time_minutes * 60,
}